code/src/esp_32.cpp

```diff
@@ -12,11 +12,14 @@ namespace r2d2::communication {
         // set the first bite of the buffer to 0
         // only the first because the first byte of an external frame is its length
         uint8_t recv[256] = {0};
-        // wait till esp is ready
-        while (!hand_shake.read()){}
+        uint8_t send[256] = {frame.length, frame.id.octets[0], frame.id.octets[1], frame.type};
+                memcpy(send+4, frame.data, frame.length );
+
+// wait till esp is ready
+        while (!hand_shake.read()){hwlib::cout << 1;}
```

Comment:
You can maybe do something other than cout << 1. Maybe you can just place continue in it.